pkg/netpol/eval/internal/k8s/netpol.go

```diff
@@ -126,9 +128,9 @@ func (np *NetworkPolicy) ruleConnections(rulePorts []netv1.NetworkPolicyPort, ds
 			if err != nil {
 				return res, err
 			}
-			if (dst == nil || isRepresentativePod(dst)) && portName != "" {
-				// adding namedPort to connectionSet in case of :
-				// - dst is nil - for general connections;
+			if (dst == nil || hasRepresentativePod(dst)) && portName != "" {
+				// adding namedPort string to connectionSet in case of :
```

Comment:
shouldn't check that exposure analysis is active?
can add more details here? why it is not added in other cases?

Reply:
updated the comment maybe it is more clear now.
k8s package doesn't have the exposure flag, but has the pods instances; so the used condition is equivelent to checking if exposure analysis is active